scipy/signal/_signaltools.py

```diff
@@ -37,6 +41,20 @@
            'filtfilt', 'decimate', 'vectorstrength']
 
 
+def _array_namespace_or_object(*args):
+    """Fall back to numpy for object arrays."""
+    try:
+        xp = array_namespace(*args)
+    except TypeError:
+        np_args = [np.asarray(a) for a in args]
+        if any(a.dtype.kind == 'O' for a in np_args):
+            xp = np
+        else:
+            # no idea what happened, just propagate the error
+            raise
+    return xp
```

Comment:
if object array support is going to be more general than `signal`, let's move this to `_array_api.py`. Why do we need to be handling object arrays though? @mdhaber can confirm, but I think in stats we are skipping tests when `SCIPY_ARRAY_API` is set if they use types which the RFC said to raise for.

Reply:
see https://github.com/scipy/scipy/blob/919f0471ede4c366cfb7266b334289b6ea4923ac/scipy/conftest.py#L167-L169